src/ucp/rndv/rndv.c

```diff
@@ -107,7 +107,9 @@ size_t ucp_rndv_rts_pack(ucp_request_t *sreq, ucp_rndv_rts_hdr_t *rndv_rts_hdr,
 
     /* Pack remote keys (which can be empty list) */
     if (UCP_DT_IS_CONTIG(sreq->send.datatype) &&
-        ucp_rndv_is_get_zcopy(sreq, worker->context)) {
+        ucp_rndv_is_get_zcopy(sreq, worker->context) &&
+        (UCP_MEM_IS_HOST(sreq->send.mem_type) ||
+         sreq->send.state.dt.dt.contig.md_map)) {
```

Comment:
minor (if any other comments): I guess better to check as:
```
(sreq->send.state.dt.dt.contig.md_map != 0)
```
since we use direct comparison to `0` if the value is not expected to be `0` and `1`, i.e. kind of flag